### DISCUSSION TITLE
How to group by arbitrary unit of time (1m, 10m, 30s, etc.)?

### DISCUSSION THREAD

**Discussion by esemeniuc** [duckdb/duckdb#9397] [Q&A] [answered]
I would like to bucket time-series data into chunks of time, eg 5m, 10s, 1d, etc. For example
```
SELECT COUNT("water_level") FROM "h2o_feet" WHERE time >= now() - 5m GROUP BY time(12m)
```
More details: https://docs.influxdata.com/influxdb/cloud/query-data/influxql/explore-data/group-by/
Does something similar exist in Duckdb?

**Comment by Tishj** [comment, marked as answer]
We dont support this directly, no

But I think maybe the `time_bucket` function in combination with windowing could replicate the behavior you're looking for?

https://duckdb.org/docs/sql/functions/timestamptz
https://duckdb.org/docs/sql/window_functions

**Reply by esemeniuc** [reply]
That looks like it! Worked for me like this:
```
select time_bucket(INTERVAL '3h', epoch_ms(time)) as bucket, round(avg("my_col")) from 'demo.csv'
group by bucket
order by bucket;
```

**Reply by Tishj** [reply]
Glad to hear it worked for you!
